### PR TITLE
JDCP-15: Define timeout for bootstrap

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

File: src/main/java/com/couchbase/client/dcp/conductor/Conductor.java [hunk 1]
```diff
@@ -85,8 +89,27 @@ public SessionState sessionState() {
...
+                                 return configProvider.start().concatWith(atLeastOneConfig).toObservable();
+                             }
+                         }
+                ).retryWhen(anyOf(TimeoutException.class)
+                        .max(Integer.MAX_VALUE)
+                        .delay(Delay.fixed(200, TimeUnit.MILLISECONDS))
+                        .doOnRetry(new Action4<Integer, Throwable, Long, TimeUnit>() {
+                            @Override
+                            public void call(Integer integer, Throwable throwable, Long aLong, TimeUnit timeUnit) {
+                                configProvider.stop().await();
```
Comment (avsej) [inline src/main/java/com/couchbase/client/dcp/conductor/Conductor.java]: @daschl what would you recommend here instead of blocking?

Comment (avsej) [reply src/main/java/com/couchbase/client/dcp/conductor/Conductor.java]: nevermind, I've decided to just add timeout and let it to bubble to the user as `BootstrapException`

File: src/main/java/com/couchbase/client/dcp/config/ClientEnvironment.java [hunk 2]
```diff
@@ -56,6 +57,11 @@
     private final String password;
 
     /**
+     * Time in milliseconds to wait before retrying bootstrap.
+     */
+    private final long bootstrapTimeout;
```
Comment (daschl) [inline src/main/java/com/couchbase/client/dcp/config/ClientEnvironment.java:63]: think this should be called connect timeout since it is used on "connect"

Comment (avsej) [reply src/main/java/com/couchbase/client/dcp/config/ClientEnvironment.java:63]: with "bootstrap" I meant that this is timeout not for whole connect action, but only for part of it, which related to waiting for initial config

**Pull request merged**